### ISSUE TITLE
Implement bulk API operations for storage-interface

### ISSUE THREAD

**Issue by harshavardhana** [minio/minio#7311]
** not using the template **

Current storage interface doesn't implement any bulk calls which can return a bulk information

Such as ReadAll() as shown here

https://github.com/minio/minio/blob/c3ca954684c685b24367da01d6fba1a92a1e2dcb/cmd/storage-interface.go#L55

The idea is that we implement a bulk ReadAllBulk() implementation which takes a list of input filenames and returns all the data necessary for each filename as a 2 dimensional byte array.

Each []byte at an index in the 2 dimensional corresponds to the index in the `paths` input argument. 

```
	ReadAllBulk(volume string, paths []string) (buf [][]byte, err error)
```

Similarly StatFile() as shown here

https://github.com/minio/minio/blob/c3ca954684c685b24367da01d6fba1a92a1e2dcb/cmd/storage-interface.go#L48

Can have a StatFileBulk() implementation

```
       StatFileBulk(volume string, paths []string) ([]FileInfo, err error) 
```

Similarly DeleteFile() as shown here

https://github.com/minio/minio/blob/c3ca954684c685b24367da01d6fba1a92a1e2dcb/cmd/storage-interface.go#L49

Can have a DeleteFileBulk() implementation

```
       DeleteFileBulk(volume string, paths []string) (err error) 
```

**Comment by EugenMayer** [comment]
Might have been implemented by https://github.com/minio/minio/issues/956 ?

**Comment by harshavardhana** [comment]
> Might have been implemented by #956 ?

@EugenMayer this issue has nothing to do with #956 - this is an internal protocol enhancement.

**Comment by harshavardhana** [comment]
Fixed by https://github.com/minio/minio/pull/7607
